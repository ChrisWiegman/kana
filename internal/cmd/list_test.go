package cmd //nolint: dupl

import (
	"bytes"
	"os/exec"
	"testing"

	"github.com/gkampitakis/go-snaps/snaps"
)

func TestList(t *testing.T) {
	t.Run("Test the default list command", func(t *testing.T) {
		cmd := exec.Command("../../build/kana", "list")
		var out bytes.Buffer
		cmd.Stdout = &out
		err := cmd.Run()
		if err != nil {
			t.Fatalf("Unexpected error: %v", err)
		}

		snaps.MatchSnapshot(t, out.String())
	})

	t.Run("Test the list command with json output", func(t *testing.T) {
		cmd := exec.Command("../../build/kana", "list", "--output-json")
		var out bytes.Buffer
		cmd.Stdout = &out
		err := cmd.Run()
		if err != nil {
			t.Fatalf("Unexpected error: %v", err)
		}

		snaps.MatchSnapshot(t, out.String())
	})
}
