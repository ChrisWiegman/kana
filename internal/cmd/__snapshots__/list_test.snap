
[TestList/Test_the_default_list_command - 1]
┌──────┬──────┬─────────┐
│ Name │ Path │ Running │
└──────┴──────┴─────────┘

---

[TestList/Test_the_list_command_with_json_output - 1]
[]

---
